x-pack/plugins/transform/public/app/hooks/use_delete_transform.tsx

```diff
@@ -4,52 +4,273 @@
+      const results: DeleteTransformEndpointResult = await api.deleteTransforms(
+        transformsInfo,
+        shouldDeleteDestIndex,
+        shouldDeleteDestIndexPattern
+      );
+      const isBulk = Object.keys(results).length > 1;
+      const successCount: Record<string, number> = {
+        transformJobDeleted: 0,
+        destIndexDeleted: 0,
+        destIndexPatternDeleted: 0,
+      };
       for (const transformId in results) {
         // hasOwnProperty check to ensure only properties on object itself, and not its prototypes
         if (results.hasOwnProperty(transformId)) {
-          if (results[transformId].success === true) {
-            toastNotifications.addSuccess(
-              i18n.translate('xpack.transform.transformList.deleteTransformSuccessMessage', {
-                defaultMessage: 'Request to delete transform {transformId} acknowledged.',
-                values: { transformId },
-              })
-            );
+          const status = results[transformId];
+          const destinationIndex = status.destinationIndex;
+
+          // if we are only deleting one modal, show the success toast messages
+          if (!isBulk) {
+            if (status.transformJobDeleted?.success) {
+              toastNotifications.addSuccess(
+                i18n.translate('xpack.transform.transformList.deleteTransformSuccessMessage', {
+                  defaultMessage: 'Request to delete transform job {transformId} acknowledged.',
+                  values: { transformId },
+                })
+              );
+            }
+            if (status.destIndexDeleted?.success) {
+              toastNotifications.addSuccess(
+                i18n.translate(
+                  'xpack.transform.deleteTransfrom.deleteAnalyticsWithIndexSuccessMessage',
+                  {
+                    defaultMessage:
+                      'Request to delete destination index {destinationIndex} acknowledged.',
+                    values: { destinationIndex },
+                  }
+                )
+              );
+            }
+            if (status.destIndexPatternDeleted?.success) {
+              toastNotifications.addSuccess(
+                i18n.translate(
+                  'xpack.transform.deleteTransfrom.deleteAnalyticsWithIndexPatternSuccessMessage',
+                  {
+                    defaultMessage:
+                      'Request to delete index pattern {destinationIndex} acknowledged.',
+                    values: { destinationIndex },
+                  }
+                )
+              );
+            }
           } else {
-            toastNotifications.addDanger(
-              i18n.translate('xpack.transform.transformList.deleteTransformErrorMessage', {
-                defaultMessage: 'An error occurred deleting the transform {transformId}',
+            Object.keys(successCount).forEach((key) => {
+              // @ts-ignore
```

Comment:
Can you please add a comment why the ignore is necessary.

Reply:
Fixed in https://github.com/elastic/kibana/pull/67922/commits/d3c37b5c0d03c2ca6e211a3ff2380154223c2f0d